harvard-lts/CURIOSity
# Make Links in Fields Clickable.

**Make Links in Fields Clickable**
* * *

**JIRA Ticket**: [LTSMAINT-757](https://jira.huit.harvard.edu/browse/LTSMAINT-757)

# Other Relevant Links (Mailing list discussion, related pull requests, etc.)
* Inspiration for the solution here: https://www.sitepoint.com/making-links/

# What does this Pull Request do?
This updates the override that SoftServe made to find all urls in the fields regardless of their placement and convert them to links...not just fields that start with a link. This leverages the regex that is built-in to URI Gem that is already included in Spotlight to find the links. I restricted it to just `http` and `https` (not `ftp` or other protocols that might show up but probably don't?) since without the restriction it was linking other words in the text that had colons in it like `series:`.

# How should this be tested?

A description of what steps someone could take to:
* Build curiosity using the `LTSMAINT-757` branch
* Go to an item that has urls in the fields (an example is https://localhost:21407/demo/catalog/47-990006049270203941_RADSCHL:1267673) 
* Confirm that urls that show up in the fields are linked to that page. In the above example, the "Notes" field has a link.

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? No
- integration tests? No

# Interested parties
@enriquediaz @dl-maura @mferrarini
